app/View/CoPeople/fields.inc

```diff
@@ -556,73 +608,140 @@
               <?php
                 $perm = $this->Permission->selfService($permissions, $e, 'Name', (!empty($n['type']) ? $n['type'] : null));
               ?>
-              <div class="field-data <?php if($perm == PermissionEnum::ReadOnly) { print ' field-data-alone'; }; ?>">
+              <div class="field-data force-wrap <?php if($perm == PermissionEnum::ReadOnly) { print ' field-data-alone'; }; ?>">
                 <?php
+                $primary_name_with_bg = $this->Utils->badgeIt(
+                  strtolower(_txt('fd.name.primary_name')),
+                  BadgeColorModeEnum::LightGray,
+                  false,
+                  true
+                );
+                $cn = $n['primary_name'] ? filter_var(generateCn($n),FILTER_SANITIZE_SPECIAL_CHARS) . "<span class='ml-1'>" . $primary_name_with_bg . "</span>"
+                                         : filter_var(generateCn($n),FILTER_SANITIZE_SPECIAL_CHARS);
                 if($perm == PermissionEnum::ReadWrite) {
-                  print $this->Html->link(generateCn($n),
+                  print $this->Html->link(
+                    $cn,
                     array('controller' => 'names',
                           'action' => 'edit',
-                          $n['id']));
+                          $n['id']),
+                    array(
+                      'escape' => false,
+                    ));
                 } elseif($perm == PermissionEnum::ReadOnly) {
-                  print filter_var(generateCn($n),FILTER_SANITIZE_SPECIAL_CHARS);
+                  print $cn;
                 }
+               ?>
+              </div>
+              <div class="field-data data-label <?php if($perm == PermissionEnum::ReadOnly) { print ' field-data-alone'; }; ?>">
+              <?php
+              $badge_list = array();
+              // OrgIdentitySource
+              if(!empty($n['SourceName']['id']) && !empty($n['SourceName']['OrgIdentity']['OrgIdentitySourceRecord']['OrgIdentitySource']['description'])) {
+                $badge_list[] = array(
+                  'order' => BadgeOrderEnum::Source,
+                  'text' => filter_var($n['SourceName']['OrgIdentity']['OrgIdentitySourceRecord']['OrgIdentitySource']['description'],FILTER_SANITIZE_SPECIAL_CHARS),
+                  'color' => BadgeColorModeEnum::Gray,
+                );
+              }
+              // Name Type
+              $badge_list[] = array(
+                'order' => BadgeOrderEnum::Status,
+                'text' => filter_var($vv_cop_name_types[ $n['type'] ],FILTER_SANITIZE_SPECIAL_CHARS),
```

Comment:
I think we'll want to continue using _txt('fd.name.primary_name') for the text of the badge here rather than the internal string reference ( $vv_cop_name_types[ $n['type'] ] ) - this will allow us to keep title case and i18n

Reply:
Done